src/shogun/lib/sg_type_traits.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Saatvik Shah
+ */
+
+#include <type_traits>
+
+#ifndef SHOGUN_SG_TYPE_TRAITS_H
+#define SHOGUN_SG_TYPE_TRAITS_H
+/**
+ * A collection of useful type traits
+ */
+
+// Checks if any one of the types matches
+// Ref: https://stackoverflow.com/a/17032517/3656081
+template <typename T, typename... Rest>
+struct is_any_of : std::false_type
```

Comment:
maybe `sg_is_any_of` ? To make clear these are shogun traits?

Reply:
would still be good to rename or at least discuss

Reply:
Yep, I'll rename it.
Apologies about the delay! Recently got back from vacation :) - I'll address all the comments this weekend.

Reply:
Hope you had a good vaccation, and happy new year :)